resources/services/memorystore/redis_instances.go

```diff
@@ -0,0 +1,352 @@
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "serial_number",
+						Description: "Serial number, as extracted from the certificate",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "sha1_fingerprint",
+						Description: "Sha1 Fingerprint of the certificate",
+						Type:        schema.TypeString,
+					},
+				},
+			},
+		},
+	}
+}
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+
+func fetchMemorystoreRedisInstances(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	c := meta.(*client.Client)
+	nextPageToken := ""
+	for {
+		call := c.Services.Redis.Projects.Locations.Instances.List("projects/" + c.ProjectId + "/locations/-").PageToken(nextPageToken)
+		list, err := c.RetryingDo(ctx, call)
+		if err != nil {
+			return diag.WrapError(err)
+		}
+		output := list.(*redis.ListInstancesResponse)
+
+		res <- output.Instances
+		if output.NextPageToken == "" {
+			break
+		}
+		nextPageToken = output.NextPageToken
+	}
+	return nil
+}
+func ResolveMemorystoreRedisInstanceID(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	// name is in the form projects/{project_id}/locations/{location_id}/instances/{instance_id}
+	parts := strings.Split(resource.Item.(*redis.Instance).Name, "/")
+	if len(parts) != 6 {
+		return diag.WrapError(
+			fmt.Errorf(
+				"name of Redis instance (%q) not in the form `projects/{project_id}/locations/{location_id}/instances/{instance_id}`",
+				resource.Item.(*redis.Instance).Name,
+			),
+		)
+	}
+	return diag.WrapError(resource.Set("id", parts[5]))
+}
+func resolveRedisInstancesMaintenancePolicy(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	instance := resource.Item.(*redis.Instance)
+	if instance.MaintenancePolicy == nil {
+		return nil
+	}
+
+	data, err := instance.MaintenancePolicy.MarshalJSON()
+	if err != nil {
+		return diag.WrapError(err)
+	}
+	return diag.WrapError(resource.Set("maintenance_policy", data))
+}
+func resolveRedisInstancesMaintenanceSchedule(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	instance := resource.Item.(*redis.Instance)
+	if instance.MaintenanceSchedule == nil {
+		return nil
+	}
+
+	data, err := instance.MaintenanceSchedule.MarshalJSON()
+	if err != nil {
+		return diag.WrapError(err)
+	}
+	return diag.WrapError(resource.Set("maintenance_schedule", data))
+}
+func resolveRedisInstancesNodes(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
```

Comment:
Sdk should marshal it automatically. If there a specific need for implementation I think it should have a comment that explains it.

Reply:
will try doing default representation